File: 2015/04/solution.rb
def solve(target)
  require 'digest'
  secret_key = File.read('secret_key')
  length = target.size
  i = 1
  while Digest::MD5.hexdigest("#{secret_key}#{i}")[0...length] != target
    i += 1
  end
  i
end

puts 'Part 1', solve('00000')
puts 'Part 2', solve('000000')

File: 2015/04/solution.php
<?php

function solve($key, $zeroes) {
    $i = 0;
    $target = str_repeat('0', $zeroes);
    while (substr(md5($key.$i), 0, $zeroes) !== $target) {
        $i++;
    }
    return $i;
}

$key = file_get_contents("secret_key");
echo "Part 1: ".solve($key, 5)."\n";
echo "Part 2: ".solve($key, 6)."\n";
